tmpl/blog/mirage-seal.md

```diff
@@ -0,0 +1,85 @@
+
+It's not strictly necessary to get someone else to sign a certificate. We can create and sign our own certificates with the `selfsign` command-line tool.  The following invocation will create a secret key in `secrets/server.key` and a public certificate for the domain `totallyradhttpsunikernel.xyz` in `secrets/server.pem`.  The certificate will be valid for 365 days, so if you choose this option, it's a good idea set a calendar reminder to renew it if the service will be up for longer than that.  The key generated will be a 2048-bit RSA key, although it's possible to create certificates valid for different lengths -- check `selfsign --help` for more information.
+
+```
+selfsign -c secrets/server.pem -k secrets/server.key -d 365 totallyradhttpsunikernel.xyz
+```
+
+We can now use this key and certificate with `mirage-seal`!  See "Packaging Up an HTTPS Site with Mirage-Seal" below.
+
+### Signed by Someone Else
+
+Although there are many entities that can sign a certificate with different processes, most have the following in common:
+
+1) you generate a request to have a certificate made for a domain
+2) the signing entity requests that you prove your ownership over that domain
+3) once verified, the signing entity generates a certificate for you
+
+#### Generating a Certificate-Signing Request
+
+No matter whom we ask to sign a certificate, we'll need to generate a certificate signing request so the signer knows what to create.  The `csr` command-line tool can do this.  The line below will generate a CSR (saved as server.csr) signed with a 2048-bit RSA key (which will be saved as server.key), for the organization "Robotic Flowers Ltd." and the common name "robotflowers.space".  For more information on `csr`, try `csr --help`.
+
+```
+csr -c server.csr -k server.key "Robotic Flowers Ltd." robotflowers.space
+```
+
+`csr` will generate a `server.csr` that contains the certificate signing request for submission elsewhere.
+
+##### Example: Gandi.net
+
+My domain is registered through the popular registrar Gandi.net, who happen to give a free TLS certificate for one year with domain registration, so I elected to have them sign a certificate for me (Gandi did not pay a promotional consideration for this mention).  Most of this process is managed through their web GUI and a fairly large chunk is automatically handled behind the scenes.  Here's how you can do it too:
+
+Log in to the web interface available through the registrar's website.  You can start the certificate signing process from the "services" tab, which exposes an "SSL" subtab.  Click that (Gandi doesn't need to know that we intend only to support TLS, not SSL).  Hit the "Get an SSL Certificate" button.  Standard SSL is fine.  Even if you're entitled to a free certificate, it will appear that you need to pay here; however at checkout, the total amount due will be 0 in your preferred currency.  Ask for a single address and, if you want to pay nothing, a valid period of 1 year.
+
```

Comment:
I didn't need to do anything with intermediate certs to get the results I expected; can you check https://dashcon2015.com with whatever browser needed them and see whether there's a problem?